src/components/activity-page/managed-interactive/managed-interactive.tsx

```diff
@@ -21,64 +22,98 @@ interface IProps {
+    const [size, setSize] = React.useState();
+
+    React.useLayoutEffect(() => {
+      setSize(target.current.getBoundingClientRect());
+    }, [target]);
+
+    useResizeObserver(target, (entry: any) => setSize(entry.contentRect));
+    return size;
+  };
+
+  const divTarget = React.useRef(null);
+  const divSize: any = useSize(divTarget);
+  const proposedHeight: number = divSize && divSize.width / aspectRatio;
+  const containerWidth: number = divSize && divSize.width;
+
+  const [ showHint, setShowHint ] = useState(false);
+  const [ hint, setHint ] = useState("");
+  const handleHintCloseClick = () => {
+    setShowHint(false);
+  };
+  const handleQuestionClick = () => {
+    setShowHint(!showHint);
+  };
+  const setNewHint = useCallback((newHint: string) => {
+    setHint(newHint);
+  }, []);
+
+  const getModalContainer = () => {
+    return document.getElementById("app") || {};
+  };
+
+  const toggleModal = () => {
+    setShowModal(!showModal);
+  };
+
+  const handleCloseModalRequest = (modalProps?: any) => {
+    if (modalProps) {
+      handleNewInteractiveState(modalProps.interactiveState);
     }
-    const url = embeddableData?.base_url || embeddableData?.url || "";
-    const linkedInteractives = (embeddable.type === "ManagedInteractive") && embeddable.linked_interactives?.length
-                                ? embeddable.linked_interactives.map(link => ({ id: link.ref_id, label: link.label }))
-                                : undefined;
-    // TODO: handle different aspect ratio methods
-    // const aspectRatioMethod = data.aspect_ratio_method ? data.aspect_ratio_method : "";
-    const nativeHeight = embeddableData?.native_height || 0;
-    const nativeWidth = embeddableData?.native_width || 0;
-    const aspectRatio = nativeHeight && nativeWidth ? nativeWidth / nativeHeight : kDefaultAspectRatio;
-
-    // cf. https://www.npmjs.com/package/@react-hook/resize-observer
-    const useSize = (target: any) => {
-      const [size, setSize] = React.useState();
-
-      React.useLayoutEffect(() => {
-        setSize(target.current.getBoundingClientRect());
-      }, [target]);
-
-      useResizeObserver(target, (entry: any) => setSize(entry.contentRect));
-      return size;
-    };
-
-    const divTarget = React.useRef(null);
-    const divSize: any = useSize(divTarget);
-    const proposedHeight: number = divSize && divSize.width / aspectRatio;
-    const containerWidth: number = divSize && divSize.width;
-
-    const [ showHint, setShowHint ] = useState(false);
-    const [ hint, setHint ] = useState("");
-    const handleHintCloseClick = () => {
-      setShowHint(false);
-    };
-    const handleQuestionClick = () => {
-      setShowHint(!showHint);
-    };
-    const setNewHint = useCallback((newHint: string) => {
-      setHint(newHint);
-    }, []);
-
-    return (
+  };
+
+  const interactiveIframe =
+    <IframeRuntime
+      url={url}
+      authoredState={embeddable.authored_state}
```

Comment:
I think we need to add back in your `embeddableAuthoredState` function:
```
  const embeddableAuthoredState = () => {
    const parsedState = embeddable.authored_state ? JSON.parse(embeddable.authored_state) : {};
    parsedState.modalSupported = true;
    return JSON.stringify(parsedState);
  };
```

And then in the props, we'll use `authoredState={embeddableAuthoredState()}`